### PR TITLE
DOC: add a skeleton and workflow for docs building

### PR DESCRIPTION
**Pull request by mrakitin**
This is the first iteration of the docs for the repo. We can do improvements via separate PRs as we go.

### CODE REVIEW

**Review by gwbischof — APPROVED** [review]

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          set -vxeo pipefail
+          # sudo apt-get update && sudo apt-get install pandoc
+          conda env list
+          pip install --upgrade pip wheel
+          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
+          pip list
+          conda list
+
+      - name: Build Docs
```
Comment (JunAishima) [inline .github/workflows/docs.yml:47]: looks like the docs are built - are they meant to be published as well?

File: docs/source/index.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    docker_image: "nsls2/linux-anvil-cos7-x86_64:latest"
+    env_name: "py39"
+    conda_env_file: "env-py39.yml"
+    conda_binary: "mamba"
+    python_version: "3.9"
+    pkg_name: ""
+    pkg_version: ""
+    extra_packages: "mamba"
+    channels: "-c conda-forge"
+
```
Comment (JunAishima) [inline docs/source/index.rst:30]: did you want to mention zenodo or other extra options (in this case, for publishing)?

**Review by JunAishima — CHANGES_REQUESTED** [review]
Should this documentation be used to replace the README.md file or replace that as well? that file has been outdated for a while.

Otherwise, this will be a valuable resource for anybody who wants to understand how this conda environment building system works, whether to add a config to build their own environment or to update the system.

File: docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+# source_suffix = ['.rst', '.md']
+source_suffix = ".rst"
+
+# The master toctree document.
+master_doc = "index"
+
+# General information about the project.
+project = "conda-pack-template"
+copyright = "2022, nsls2-conda-envs"
+author = "nsls2-conda-envs"
```
[the remark this answers is not in the document]
Comment (mrakitin) [reply docs/source/conf.py:73]: Good point, I'll correct it. The values were added automatically by the cookiecutter.

File: docs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+- ``conda_binary`` is the conda binary --- ``conda`` (default) or ``mamba``
+  (for faster execution)
+- ``pkg_name`` is the conda package to install (can be an empty string)
+- ``pkg_version`` is the version of the ``pkg_name``
+- ``extra_packages`` lists the extra conda packages to install along with
+  ``pkg_name``
+- ``extra_cmd_before_install`` (not listed in the above example) is used to
+  specify the command to run in Docker before conda environment installation
+- ``extra_cmd_after_install`` (not listed in the above example) is used to
+  specify the command to run in Docker after conda environment installation
```
[the remark this answers is not in the document]
Comment (mrakitin) [reply docs/source/index.rst:54]: I'll add a couple of examples.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (mrakitin) [reply .github/workflows/docs.yml:47]: Yes, I'll address it via a separate PR -- just wanted to make sure we cover the basics first.

File: docs/source/index.rst (hunk 2 above)
Comment (mrakitin) [reply docs/source/index.rst:30]: Zenodo publishing is pretty experimental for now in this repo. We need to update the code in `zenodo.py` to allow publishing reliably. I'll need to combine the code with https://github.com/nsls2-conda-envs/playground/blob/ci-upload-nsls2-collection-2022-2.4/test-upload.py. Once the code is added, we can update the documentation.

**Review by JunAishima — APPROVED** [review]

**Comment by mrakitin** [comment]
I changed the sphinx theme to https://cloud-sptheme.readthedocs.io/en/latest/ (similar to https://github.com/NSLS-II/sirepo-bluesky/pull/80). The built docs will look like:

image

**Comment by mrakitin** [comment]
@JunAishima, @gwbischof, ready for your review. The built docs are available in https://github.com/nsls2-conda-envs/conda-pack-template/actions/runs/3085767320 if you'd like to check the rendered version.

**Comment by mrakitin** [comment]
OK, I am going to merge it given one review (that I dismissed to solve the CI/STY issues.)

**Comment by mrakitin** [comment]
The docs are live now: https://nsls2-conda-envs.github.io/conda-pack-template (implemented via https://github.com/nsls2-conda-envs/conda-pack-template/commit/1ae01fa34c1a666e089e539ed92ba8f8e0184b28 and the repo settings).

The https://github.com/nsls2-conda-envs/conda-pack-template/commit/0ff2ad364ad85d2b96ebea6cf2c8b6eb286b75e3 commit updates the `README.md` file.

The next step is to rename the repo and apply the changes to all env-repos.

**Pull request merged**